app/javascript/components/supplemental_application/actions/preferenceActionCreators.js

```diff
@@ -5,10 +5,15 @@ import ACTIONS from 'context/actions'
  * Returns a promise that resolves to true if the save was successful, false otherwise.
  */
 
-export const updateSavedPreference = async (dispatch, preferenceIndex, formApplicationValues) => {
+export const updateSavedPreference = async (
+  dispatch,
+  preferenceIndexToUpdate,
+  preferenceIndexToClose,
+  formApplicationValues
+) => {
```

Comment:
question: I'm curious about what this change addresses, maybe a bug? I'm still not familiar with how state management works in LAP.

Reply:
It stems from the requirement in this ticket where they no longer want veteran preference to be editable but the non veteran preference should remain editable from the veteran panel. The preference that i need to save is not always the current index, but the preference i need to close will be the current index.

https://github.com/SFDigitalServices/sf-dahlia-lap/blob/f300f8e74a116ef8474a62150498046983c9e9b0/app/javascript/components/supplemental_application/sections/preferences/Panel.js#L74-L82

![Screenshot 2024-03-26 at 12 06 31 PM](https://github.com/SFDigitalServices/sf-dahlia-lap/assets/135058064/b059820d-9cf3-4166-82e0-abcc9645b2cd)